arangosh/Shell/V8ClientConnection.cpp

```diff
@@ -1863,7 +1863,11 @@ v8::Local<v8::Value> V8ClientConnection::requestDataRaw(
   req->header.parseArangoPath(location.toString());
   for (auto& pair : headerFields) {
     if (boost::iequals(StaticStrings::ContentTypeHeader, pair.first)) {
-      req->header.contentType(fu::ContentType::Custom);
+      if (pair.second == StaticStrings::MimeTypeJsonNoEncoding) {
+        req->header.contentType(fu::ContentType::Json);
+      } else {
+        req->header.contentType(fu::ContentType::Custom);
+      }
     } else if (boost::iequals(StaticStrings::Accept, pair.first)) {
       req->header.acceptType(fu::ContentType::Custom);
     }
```

Comment:
I don't see how this will help much.
For both headers `Content-Type` and `Accept` we will do something special here (i.e. set the `_contentType` or `_acceptType` in `req->header`).
However, after we do this we unconditionally call `req->header.addMeta(...)`, which for these headers will unconditionally overwrite `_contentType` or `_acceptType`.

If this is true, we can remove the whole special handling here, and simply call `req->header.addMeta(...)` here without _any_ special handling.